collect_app/src/main/java/org/bcss/collect/naxa/login/LoginModelImpl.java

```diff
@@ -66,27 +67,19 @@ public void onNext(FCMParameter fcmParameter) {
                     @Override
                     public void onError(Throwable e) {
                         Timber.e(e);
-                         if (e instanceof HttpException) {
-                            HttpException httpException = (HttpException) e;
-                            int statusCode = httpException.response().code();
-                            switch (statusCode) {
-                                case 400:
-                                    String parsedErrorMessage = APIErrorUtils.getNonFieldError(httpException);
-                                    onLoginFinishedListener.onError(parsedErrorMessage);
-                                    break;
-                                default:
-                                    onLoginFinishedListener.onError("Server returned " + statusCode);
-                            }
-                        } else if (e instanceof FirebaseTokenException) {
-                            onLoginFinishedListener.fcmTokenError();
+                        String errorMessage = e.getMessage();
+                        if (e instanceof RetrofitException) {
+                            RetrofitException retrofitException = (RetrofitException) e;
+                            boolean hasErrorBody = retrofitException.getResponse().errorBody() != null;
+                            errorMessage = hasErrorBody ? retrofitException.getMessage() : retrofitException.getKind().getMessage();
                         } else if (e instanceof SSLException) {
-                            onLoginFinishedListener.onError("An SSL exception occurred");
-                        } else {
-                            onLoginFinishedListener.onError("Generic error occurred: " + e.getMessage());
-
+                            errorMessage = "A SSL exception occurred";
+                        }
+                        if (e instanceof FirebaseTokenException) {
+                            onLoginFinishedListener.fcmTokenError();
+                        }else {
+                            onLoginFinishedListener.onError(errorMessage);
```

Comment:
```
 if (e instanceof FirebaseTokenException) {

                             onLoginFinishedListener.fcmTokenError();
                        }else {
                            onLoginFinishedListener.onError(errorMessage);
                          }
```
Replace this code with this

` onLoginFinishedListener.onError(errorMessage);`

Reply:
Yes ! nice catch